src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
+        int oldCapacity = array.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
```

Comment:
It is better to extract multiplication into private static class variable and change code into 
```suggestion
            int oldCapacity = array.length;
            double multiplicator = 1;
            if ((oldCapacity > size || oldCapacity == size) || array != EMPTY_ARRAYDATA) {
            int newCapacity = (int) (oldCapacity * MULTIPLICATOR + 1);
            return array = Arrays.copyOf(array, newCapacity);
        }
```